src/components/VotingSystem/VotingSystem.jsx

```diff
@@ -0,0 +1,43 @@
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import ThumbDownIcon from "@mui/icons-material/ThumbDown";
+import { Box, Button } from "@mui/material";
+import { patchArticleById } from "../api";
+
+export const VotingSystem = ({ id, setArticle }) => {
+  const handleUpVote = () => {
+    const newVote = 1;
+    patchArticleById(id, newVote)
+      .then((response) => {
+        setArticle((currArticleData) => {
+          return { ...currArticleData, votes: response.votes };
```

Comment:
We would like optimistic rendering here. Please set the article changes before the api request and undo them in the catch block 🙂 

Overall its good logic though 👍